server/controllers/usersController.js

```diff
@@ -55,47 +57,54 @@ export default class Users {
       where: {
         email
       }
-    })
-      .then((existing) => {
-        if (existing) {
-          return res.status(403).send({
-            status: 'Forbidden',
-            message: 'Email already exists.',
+    }).then((existingUser) => {
+      if (existingUser) {
+        return res.status(409)
+          .json({
+            status: 'fail',
+            message: 'Email already exist',
           });
-        }
-        return db.User
-          .create({
-            fullname,
-            username,
-            email,
-            password: bcrypt.hashSync(req.body.password, 10),
-          })
-          .then((newUser) => {
-            const token = jwt.sign({ id: newUser.id }, process.env.MY_SECRET, { expiresIn: '24h' });
-            res.status(201)
-              .json({
-                status: 'success',
-                message: 'Account created',
-                user: {
-                  username: newUser.username,
-                  fullname: newUser.fullname,
-                  email: newUser.email,
-                },
-                token
-              });
-          })
-          .catch(error => res.status(400).send(error.message));
-      }).catch(error => res.status(400).send(error.message));
+      }
+      return db.User
+        .create({
+          fullname,
+          username,
+          email,
+          password: bcrypt.hashSync(req.body.password, 10),
+        })
+        .then((newUser) => {
+          const token = jwt.sign(
```

Comment:
### Critical
Token generation is done twice in this class and that negates the DRY principle. You can fix this by abstracting a frequently used functionality into a function/class method, let's call the class`utility` or `helper`. Token concerns such as `generateToken`, `getToken` and `verifyToken` should all be members of this class.

Reply:
Fixed